helix-stdx/src/lib.rs

```diff
@@ -1,4 +1,31 @@
+use std::path::Path;
+
 pub mod env;
 pub mod faccess;
 pub mod path;
 pub mod rope;
+
+#[cfg(unix)]
+pub fn get_hardlink_count(p: &Path) -> std::io::Result<u64> {
+    use std::os::unix::fs::MetadataExt;
+    let metadata = p.metadata()?;
+    Ok(metadata.nlink())
+}
+
+#[cfg(windows)]
+pub fn get_hardlink_count(p: &Path) -> std::io::Result<u64> {
+    use std::os::windows::io::IntoRawHandle;
+    use windows_sys::Win32::Storage::FileSystem::{
+        GetFileInformationByHandle, BY_HANDLE_FILE_INFORMATION,
+    };
+
+    let file = std::fs::File::open(p)?;
+    let handle = file.into_raw_handle() as isize;
```

Comment:
Thus leaks the file handle as it never gets closed.

You need to use as_raw_handle instead:

 https://doc.rust-lang.org/std/os/windows/io/trait.AsRawHandle.html#tymethod.as_raw_handle